arm-datalake-store/account/2016-11-01/swagger/account.json

```diff
@@ -0,0 +1,1311 @@
+          "$ref": "#/definitions/FirewallRuleProperties",
+          "description": "the properties of the firewall rule."
+        }
+      },
+      "description": "Data Lake Store firewall rule information"
+    },
+	"TrustedIdProviderProperties": {
+      "properties": {
+        "idProvider": {
+          "type": "string",
+          "description": "The URL of this trusted identity provider"
+        }
+      },
+      "description": "Data Lake Store trusted identity provider properties information"
+    },
+	"TrustedIdProvider": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "the trusted identity provider's name."
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the namespace and type of the trusted identity provider."
+        },
+        "id": {
+          "type": "string",
+          "description": "the trusted identity provider's full ID."
+        },
+        "properties": {
+          "$ref": "#/definitions/TrustedIdProviderProperties",
+          "description": "the properties of the trusted identity provider."
+        }
+      },
+      "description": "Data Lake Store firewall rule information"
+    },
+	"DataLakeStoreTrustedIdProviderListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/TrustedIdProvider"
+          },
+          "description": "the results of the list operation"
+        },
+        "nextLink": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the link (url) to the next page of results."
+        },
+        "count": {
+          "type": "integer",
+          "readOnly": true,
+          "format": "int64",
+          "description": "the total count of results that are available, but might not be returned in the current page."
+        }
+      },
+      "description": "Data Lake Store trusted identity provider list information."
+    },
+    "DataLakeStoreFirewallRuleListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/FirewallRule"
+          },
+          "description": "the results of the list operation"
+        },
+        "nextLink": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the link (url) to the next page of results."
+        },
+        "count": {
+          "type": "integer",
+          "readOnly": true,
+          "format": "int64",
+          "description": "the total count of results that are available, but might not be returned in the current page."
+        }
+      },
+      "description": "Data Lake Store firewall rule list information."
+    },
+    "EncryptionIdentity": {
+      "properties": {
+        "type": {
+          "type": "string",
```

Comment:
You don't have to pass in encryption information, but if you do certain things are required. I will add explicit requirements further down the property chain.